alpheios_nemo_ui/data/assets/js/infinite-scroll.js

```diff
@@ -0,0 +1,97 @@
+var nextPassage = {};
+var isloading = false;
+var isInit = false;
+var uploadOneMore = false;
+var uploadOnemoreCheck  = null;
+
+function preloadNext() {
+    if ($('#current-passage').length > 0) {
+        var deferred = $.Deferred();
+
+        let currentSubrefData = $('#current-passage').data();
+        if (!nextPassage.start) {
+            nextPassage.start = currentSubrefData.subreference.split('-')[0]
+        }
+        isloading = true
+        $.getJSON( `/nextpassage/${currentSubrefData.objectid}/${currentSubrefData.subreference}`)
+          .done(function( urlData ) {
+            deferred.notify(urlData)
+
+            $.getJSON(`${urlData.url}/json`)
+            .done(function(data) {
+                let nextArr = urlData.next.split('-');
+                Object.assign(nextPassage, {
+                    data: data,
+                    next: urlData.next,
+                    subreference: currentSubrefData.subreference,
+                    finalRef: nextPassage.start + '-' + nextArr[nextArr.length-1],
+                    nextRefUrl: urlData.nextRefUrl
+                })
+                
+                console.log('Retrieved next data', urlData.url)
+                isloading = false
+                deferred.resolve(nextPassage);
+            })
+            .fail(function(error) {
+              console.log( "error" , error);
+            });
+          })
+        return deferred.promise();
+    }
+}
+
+function uploadNext() {
+    if (nextPassage && nextPassage.data) {
+        if (nextPassage.data.new_level) {
+            $('#article-entry').append(`<div class="newlevel">${nextPassage.data.new_level}</div>`)
+        }
+        $('#article-entry').append(nextPassage.data.text_passage);
+        $('#current-passage').data('subreference', nextPassage.next);           
+        $('#current-passage').text(nextPassage.finalRef);
+
+        $('#next-passage').data('prev', nextPassage.data.next);
+        $('#next-passage').prop('href', nextPassage.data.next);
+        if (!isInit) {
+            preloadNext();
+        }
+    }
+}
+
+function initPassages() {
+    if ($(window).scrollTop() + window.innerHeight + 10 > $(document).height()) {
```

Comment:
I think if this test fails, i.e if the original passages fill the entire page, then next passage never gets loaded and we never get initialized properly.  I haven't fully traced this through but it seems only to be working to me if this test is true to start (for example if I load http://localhost:5000/text/urn:cts:latinLit:phi0474.phi045.perseus-lat1/passage/1.1-1.2 which is a small passage range and doesn't fill my entire screen)

Reply:
Oh, you are right - it won't be able to change the isInit flag to false.
I will add changes.

Reply:
I have fixed :) Thank you for your attention:)